core/src/main/java/org/apache/iceberg/StaticTableOperations.java

```diff
@@ -29,24 +29,37 @@
  * and cannot be used to create or delete files.
  */
 public class StaticTableOperations implements TableOperations {
-  private final TableMetadata staticMetadata;
+  private TableMetadata staticMetadata;
+  private final String metadataFileLocation;
   private final FileIO io;
+  private final LocationProvider locationProvider;
 
   /**
    * Creates a StaticTableOperations tied to a specific static version of the TableMetadata
    */
   public StaticTableOperations(String metadataFileLocation, FileIO io) {
+    this(metadataFileLocation, io, null);
+  }
+
+  public StaticTableOperations(String metadataFileLocation, FileIO io, LocationProvider locationProvider) {
     this.io = io;
-    this.staticMetadata = TableMetadataParser.read(io, metadataFileLocation);
+    this.metadataFileLocation = metadataFileLocation;
+    this.locationProvider = locationProvider;
   }
 
   @Override
   public TableMetadata current() {
+    if (staticMetadata == null) {
+      staticMetadata = TableMetadataParser.read(io, metadataFileLocation);
+    }
     return staticMetadata;
   }
 
   @Override
   public TableMetadata refresh() {
+    if (staticMetadata == null) {
```

Comment:
Do we need to refresh static tables?

Reply:
Also, I was thinking that not refreshing the metadata on calling `refresh()` might come as a surprise for users. Do we want to either throw unsupported operation exc (probably not preferable), or at least add some short javadoc explaining why the refresh() works differently here than for the other table operations? We might even simply call `current()` here to make it clear it's the same behaviour. Let me know what you guys think

Reply:
Comment is needed, but I think not doing anything is the best option.
Calling `current()` calls metadata parsing, which might be unnecessary if the metadata is not used later.

Reply:
Added a comment to the refresh method.

Reply:
We have a couple of negative test cases that tries to modify StaticTables. Obviously, the operation will fail with an exception during commit(), but before it does it tries to create a snapshot that calls refresh(). If we return null, we will run into a nullpointer exception.